WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
         "ST_s-channel_4f_leptonDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=3.34),
+            "CrossSection" : XSValues(
+                XS_13TeV=3.36, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25n (NLO)",
+                XS_2016=3.365, XSSource_2016="XSDB (unknown)",
+                XS_2017=3.74,  XSSource_2017="XSDB (unknown)",
+                XS_2018=3.740, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ST_s-channel_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=10.32),
+            "CrossSection" : XSValues(
+                XS_13TeV=10.12, XSSource_13TeV="XSDB (unknown)",
+            ),
         },
         "ST_t-channel_top_4f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=136.02),
+            "CrossSection" : XSValues(
+                XS_13TeV=136.02, XSSource_13TeV="Unknown",
+                XS_2017=113.3,   XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_t-channel_top_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=136.02),
+            "CrossSection" : XSValues(
+                XS_13TeV=136.02, XSSource_13TeV="Unknown",
+                XS_2017=113.3,   XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_t-channel_antitop_4f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=80.95),
+            "CrossSection" : XSValues(
+                XS_13TeV=80.95, XSSource_13TeV="Unknown",
+                XS_2017=67.91,  XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_t-channel_antitop_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=80.95),
+            "CrossSection" : XSValues(
+                XS_13TeV=80.95, XSSource_13TeV="Unknown",
+                XS_2017=67.91,  XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_tW_antitop_5f_NoFullyHadronicDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=19.4674),
+            "CrossSection" : XSValues(
+                XS_13TeV=19.4674, XSSource_13TeV="Unknown",
+                XS_2016=38.06,    XSSource_2016="XSDB (NLO)",
+                XS_2017=34.97,    XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_tW_top_5f_NoFullyHadronicDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=19.4674),
+            "CrossSection" : XSValues(
+                XS_13TeV=19.4674, XSSource_13TeV="Unknown",
+                XS_2016=38.09,    XSSource_2016="XSDB (NLO)",
+                XS_2017=34.91,    XSSource_2017="XSDB (NLO)",
+            ),
         },
         "ST_tW_antitop_5f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=35.6),
+            "CrossSection" : XSValues(
+                XS_13TeV=35.6, XSSource_13TeV="Unknown",
+                XS_2016=38.06, XSSource_2016="XSDB (NLO)",
```

Comment:
i don't believe that inclusiveDecays can have the same xsec as NoFullyHadronicDecays
one of these values is wrong

Reply:
consider comment at https://github.com/TreeMaker/TreeMaker/pull/505/files#diff-cc5b25d546afd2a07d4cc2600dcf2cedR830

Reply:
In other words you want me to multiply the XS_2016 and XS_2017 values by the formula in that comment for the NoFullyHadronicDecays samples? Where does the 0.108 factor come from? Or do you just want me to confirm all of these values using the GenXSecAnalyzer?

Reply:
0.108 = Br(W->ℓν)

Can you run GenXsecAnalyzer and check for consistency with the expectation from branching fraction?

Reply:
I don't know of an option like that for the GenXSecAnalyzer. I re-ran it for all of the single top samples to confirm what is in the XSDB. They matched.

I'm going to run jobs to get the 2018 values as I realize they are missing. What would you like me to do about the NoFullyHadronicDecays samples?

Reply:
I just meant compare the GenXsecAnalyzer values for Inclusive and NoFullyHadronicDecays. If they come out the same, then I guess we have to multiply the BR by hand.

Reply:
They came out the same for 2016 and 2017. I'm running 2018 now, but I don't expect there to be anything different.

Just to confirm, you want me to scale the NoFullyHadronicDecays like `XS*(1-(1-3*0.108)^2)`? I can then make a note of this in the XS_Source section so that we don't have to go through this exercise again.

Reply:
Yes

Reply:
Let me know what you think of this implementation.